src/tests/math/test_stdlib_math.fypp

```diff
@@ -203,6 +210,69 @@ contains
 #:endif
 
     end subroutine test_clip_rqp_bounds
+    
+    #:for k1 in CMPLX_KINDS
```

Comment:
Do you plan to only add tests for zeros arrays?

Reply:
my mistake: you also test for one non-zero value.

Reply:
I think a few more tests are needed. 

For convenience these could use vector arguments, which would cover a range of cases at once and exploit the `elemental` nature of the routines.

As one suggestion - define `theta` as an angle taking 100 evenly spaced values on the unit circle. Compute the associated complex numbers, and then check that the results of `arg*` are an evenly spaced sequence with the expected range, and accurate to the expected precision.

Reply:
You might consider modifying this code, which implements a test like above (double precision only, but with obvious fypp changes). 
```fortran
program arg_routines
    use stdlib_kinds, only: sp, dp, qp
    use stdlib_math, only: arg, argd, argpi
    implicit none

    integer, parameter :: N = 1000
    real(dp), parameter :: pi = acos(-1.0_dp), tol = 1000.0_dp * epsilon(1.0_dp)
    real(dp) :: theta(N), results(N), expected_results(N)
    complex(dp) :: z(N)
    integer:: i

    theta = [ ((i-1)*1.0_dp/(N+1), i = 1, N) ] ! Fraction of one circular rotation

    ! Complex numbers with evenly spaced arguments (and varying modulus)
    z = cmplx(cos(theta*2*pi), sin(theta*2*pi), kind=dp)*(theta + tol)

    ! Argd should range from 0->180, then jump to -180->0
    ! The value 
    results = argd(z)
    expected_results = theta * 360_dp
    where(expected_results > 180.0_dp) expected_results = expected_results - 360.0_dp
    if(all(abs(results - expected_results) < tol)) then
        print*, "PASS"
    else
        print*, 'FAIL'
    end if

    ! Arg should range from 0->pi, then jump to -pi->0
    results = arg(z)
    expected_results = expected_results/180.0_dp * pi
    if(all(abs(results - expected_results) < tol)) then
        print*, "PASS"
    else
        print*, 'FAIL'
    end if

    ! Argpi should range from 0->1, then jump to -1->0
    results = argpi(z)
    expected_results = expected_results/pi
    if(all(abs(results - expected_results) < tol)) then
        print*, "PASS"
    else
        print*, 'FAIL'
    end if

end program
```